### PR TITLE
Draft: Governance

### PR DESCRIPTION
**Pull request by knarz**
This draft will provide background and description for aeternity's governance. This currently focuses on the background, because I am of the opinion that giving a thorough explanation for the different decisions is much more important in this case, than actual technical decisions.

Things to be added:
- specification for replay protection based on commitments on improvement proposals
- specification for delegation process
- description of the AEIP process (although it will be similar to EIP/BIP process)
- justifications and explanations for rejection of on chain amendments

Nonetheless, I'd like to start the discussion around this.

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+#### Proof of Work
+
+- allows miners to vote
+
+#### Proof of Stake
+
+- dash proposals gather ~25% masternode (~4500) votes
+- even if we allow stake holders and miners to contribute (via pos and pow
+  respectively) then there are still more groups of people: node operators,
+  who validate the chains, and end users, both of which have who have no good
```
Comment (lucafavatella) [inline GOVERNANCE.md]: s/both of which have who have/both of which have

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+Using votes to decide on forks could be seen as detrimental to this goal, since
+a vote has loser by design—if you are not voting for me, you're voting against
+me.
+
+
+#### Exit costs
+
+One of our goal is to make exit costs low, i.e. allow users to easily abandon
+undesirable forks. But it can be argued that if people have very low exit costs
+for a given system, then they incentive to vote according to their true beliefs,
```
Comment (lucafavatella) [inline GOVERNANCE.md]: s/they/their

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+
+### Constitution
+
+- what is allowed to be decided on
+- rules on how to change rules are informed by expectations of what the outcomes
+  should/could be
+
+
+### Automatic protocol amendments
```
Comment (lucafavatella) [inline GOVERNANCE.md]: The previous paragraph refers to "constitution". This paragraph has in title "protocol" though in its body refers again to "constitution", so I understand "protocol" here may be replaced with "constitution".

Comment (knarz) [reply GOVERNANCE.md]: No, these are two distinct issues. I'll push my latest changes in a bit

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+relevant consensus rules in a language for a virtual machine every participant
+has to run or by using a language for these change that can be transpiled easily.
+
+Now if we assume that we have such a language, then the next step is to verify
+that the given piece of code adheres to the constitution and does not violate
+any of its invariants. This could either be done via manual, human review or
+formal methods, i.e. having both the constitution and the code be formally
+verifiable. The problem is, that this is most likely a gross oversimplification
+because software engineering is never as simple as just writing a single piece
+of code and have it just magically work, especially not
```
Comment (cytadela8) [inline GOVERNANCE.md]: This sentence is confusing, it's seams to and abruptly. At least it's missing a `.` at the end.

**Pull request closed without merging**